Dockerfile

```diff
@@ -1,29 +1,37 @@
-FROM python:3.8
+# Stage 1: Start from the official Python-3.8 image: https://hub.docker.com/_/python/
+FROM python:3.8 AS base
 
 # system maintenance
 RUN apt-get update
 
 # install dependencies needed for setting up R
 RUN apt-get install -y lsb-release dirmngr gnupg apt-transport-https ca-certificates software-properties-common
 RUN apt-get install -y libharfbuzz-dev libfribidi-dev
-RUN apt-get -y install libcurl4-gnutls-dev
-
-# set up the key for adding the R repo
-RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 95C0FAF38DB3CCAD0C080A7BDC78B2DDEABC47B7
-
-# add the correct Linux R repo
-RUN add-apt-repository 'deb https://cloud.r-project.org/bin/linux/debian bullseye-cran40/'
+RUN apt-get install -y libcurl4-openssl-dev libssl-dev
+RUN apt-get install -y libcurl4-gnutls-dev
+RUN apt-get install -y libopenblas-base libgit2-dev
 
+# add the correct Debian R repo
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys '95C0FAF38DB3CCAD0C080A7BDC78B2DDEABC47B7'
+RUN add-apt-repository 'deb http://cloud.r-project.org/bin/linux/debian bullseye-cran40/'
 # re-update based on previous setup
-RUN apt-get update && apt-get -y upgrade
+RUN apt-get update && apt -y upgrade
 
-# install gcc and R
-RUN apt-get install -y gcc r-base
+# install R
+ARG rversion=4.0.4-1
+RUN apt-get install -y \
+    r-base-core=${rversion} \
+    r-base-dev=${rversion} \
+    r-base-html=${rversion} \
+    r-doc-html=${rversion}
```

Comment:
Does the Python 3.8 image require us to explicitly specify all of these now?

Reply:
Yeah, for native M1 support I had to do this. If I just did `apt-get install -y r-base` it would always fail stating that there was a version mismatch for the requirements of `r-base`.